### ISSUE TITLE
References should be linked to from Product Docs ToC

### ISSUE THREAD

**Issue by leggetter** [Nexmo/nexmo-developer#156]
## Description

Within the LHS table of contents in Product documentation I would like to also see a link to the API Reference documentation.

<img width="310" alt="screen shot 2017-07-03 at 12 00 00" src="https://user-images.githubusercontent.com/328367/27790136-2e6e92c8-5fe7-11e7-9eb9-1b881dd5f26c.png">

The potential problem with this is that the reader may be confused when clicking the link results in them being taken to a whole new section of the documentation with a different LHS ToC.

Could/should the API Reference for each product be within the product documentation? Should https://developer.nexmo.com/api/voice actually be https://developer.nexmo.com/voice/api-reference? If so, how would the navigation system work?

## API Reference -> Reference

Please also note that some products may have multiple types of reference e.g. a REST API Reference and an SDK API Reference. So, should we update "API Reference" to be "Reference"?

## Suggested Plan

Short-term: Maybe a link in Product/LHS ToC that has a small visual indicator that the link takes you out of the current section?

Longer-term: We may need to rethink the LHS ToC to support References also being present.

**Comment by adambutler** [comment]
The problem I see is that:

1. The API Reference pages will diverge from documentation over time. These are very deliberately distinct sections with difference purposes and likely different interfaces in the future.

2. SMS & Verify have multiple reference pages (at this time at least):

    **SMS:**

    - SMS API
    - Alerts API - Subscribing
    - Alerts API - Sending
    - 2FA

    **Verify:**

    - Verify API
    - Verify API - Custom templates

    Would we link to the canonical one or list them all?

3. "Could/should the API Reference for each product be within the product documentation?"

    I think we can either link to the canonical page with a visual cue or hold this off until the API reference pages have been reworked.

4. "So, should we update "API Reference" to be "Reference"?"

    I'm fond of the way that Stripe has done their [API Reference](https://stripe.com/docs/api) and believe that we can and should consolidate the REST & SDK references together. Therefor I think it should remain under its current route and title. This is a huge discussion itself and the work to make this happen would be considerable. I'd encourage this to be opened as its own issue.

5. LHS ToC? Can we refrain from using unnecessary abbreviations.

    a. Left hand side?
    b. ToC?

---

In summary I propose:

**Short term**

That we add an "API Reference" link to each of the products linking to the canonical API reference page for that product.

**Longer term**

That the API reference pages be reworked to be consolidated and have both REST and SDK references. This is just a proposal and a potential distraction, a further discussion should be had once we have completed the current milestone.

**Comment by leggetter** [comment]
@adambutler your short term proposal SGTM 😜

LHS = Left Hand Side
ToC = Table of Contents
